Objects/Converters/ConverterNavisworks/ConverterNavisworks/ConverterNavisworks.Properties.cs

```diff
@@ -12,40 +12,27 @@ namespace Objects.Converter.Navisworks
 {
   public partial class ConverterNavisworks
   {
-    internal static Base GetPropertiesBase(ModelItem element, ref Base @base)
```

Comment:
I know this was discussed in Discord birefly... (maybe I should have weighed in)

This is not to stop you from merging this at all... But if this method has _side-effects_ on `@base`... the `ref` was a good indicator that _something_ could happen to that object.

@JR-Morgan may have some opinions on this.

Reply:
To be discussed with the team what our "ref stance" is. No particular opinions other than it seems like a nice hint to add it when there are side-effects, but the reality is that code-wise it does nothing.

Reply:
As for communicating side effects. I agree, C# language does a poor job of allowing you to enforce this.
Other languages have `const` correctness, or object immutability rules
Some would say the way to go is to have getter only interfaces,
Some would say that you should just document when you mutate objects and/or make it obvious by the method name.

I don't have much wisdom to offer on this, other than, I don't think the `ref` keyword is the way to go.